packages/validator/core/validator-functions.ts

```diff
@@ -7,42 +7,74 @@ import { Validators } from './validator-names';
  * @param validators - names of validation logic to be applied
  * @returns errors - array of errors `ValidationError`
  */
-export function validate(value: string, validators: string[]): ValidationError[] {
+export function validate(value: string, validators: ValidatorRules): ValidationError[] {
 	return validators
-		.map((validator) => validator.replace('data-', ''))
-		.map((attribute): ValidationError | null => {
+		.map((validator) => {
+			if (typeof validator === 'string') {
+				return { attribute: validator.replace('data-', ''), category: 'error' };
+			}
+
+			return {
+				attribute: validator.validator.replace('data-', ''),
+				category: validator.category || 'error',
+			};
+		})
+		.map(({ attribute, category }): ValidationError | null => {
 			// TODO: implement dynamic import of function depending on validators
 			const split = attribute.split(':');
 			const validator = split[0];
 			const limitStr = split[1];
 			const limit = parseInt(limitStr || '0', 10);
 
 			if (validator === Validators.min()) {
-				return validateMin(value, limit);
+				const val = validateMin(value, limit);
+				if (val) {
```

Comment:
Generally, you want to minimize the if-within-an-if to reduce the function complexity.

I would add an optional parameter in the validation functions, like so: `validateMin(value, limit, category)` then add the `category` prop in the object they return :)

Reply:
Right sure! I am just trying out different things and seeing what will work best!